### PR TITLE
Fix incorrect severity syslog messages

### PR DESCRIPTION
**Pull request by drTr0jan**
Fix `logging` for send syslog messages with correct severity.
Before the proposal patch syslog messages always were sent with `NOTICE` severity. #12058 

Does this patch affects other modules?

### CODE REVIEW

**Review by markan — COMMENTED** [review]
Thank you for the fix.  This looks clean and correct. 

It looks like we didn't test syslog specifically, explaining how this slipped through.
I'd love to have a test here, but it's not immediately obvious how to verify syslog is inserting the correct PRI field or otherwise check that the Severity field is set as intended.

**Comment by markan** [comment]
Also could you add a changelog entry: https://github.com/hashicorp/consul/blob/main/docs/contributing/add-a-changelog-entry.md

**Comment by drTr0jan** [comment]
@markan, I've done a changelog entry. But unfortunately I cann't make a test to syslog too.

**Comment by eculver** [comment]
I added tests for the syslog wrapper that is now being used.

File: logging/syslog.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ func (s *SyslogWrapper) Write(p []byte) (int, error) {
 		y := bytes.IndexByte(p[x:], ']')
 		if y >= 0 {
 			level = string(p[x+1 : x+y])
-			afterLevel = p[x+y+2:]
+			afterLevel = bytes.TrimLeft(p[x+y+2:], " ")
```
Comment (eculver) [inline logging/syslog.go:37]: I'm not sure we need this. I'm curious if you were seeing messages with extra spaces between the level generated somewhere? If so, could you tell us where? 

I made sure the test cases covered this little bit, so if you decide to remove it, you'll need to update the tests to remove those extra spaces.

Comment (drTr0jan) [reply logging/syslog.go:37]: Nope, I was seeing messages without extra spaces between the level generated. I was seeing messages with extra spaces before the log message text (an MSG field RFC3164/5424). In other words, all messages contain MSG field with preceding extra space.

**Review by markan — APPROVED** [review]

**Comment by drTr0jan** [comment]
What about merge?

**Comment by eculver** [comment]
Looking into the test failures. Once we resolve these, we can get this merged.

**Review by eculver — APPROVED** [review]

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/640030.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 319a0a08cdc57a9d771cf413a12ec1b668663f91 onto `release/1.12.x` succeeded!

**Pull request merged**